plugins/bidi/plugin.js

```diff
@@ -236,8 +236,14 @@
 
 			var lang = editor.lang.bidi;
 
-			addButtonCommand( 'BidiLtr', lang.ltr, 'bidiltr', bidiCommand( 'ltr' ), 10 );
-			addButtonCommand( 'BidiRtl', lang.rtl, 'bidirtl', bidiCommand( 'rtl' ), 20 );
+			if (editor.lang.dir=="rtl") {
+				// reversed "order" of toolbar icons when editor language is RTL so would look the right way
+				addButtonCommand( 'BidiRtl', lang.rtl, 'bidirtl', bidiCommand( 'rtl' ), 10 );
```

Comment:
This code should be simplified. You don't have to copy and paste both calls to `addButtonCommand`, instead you could just use a ternary operator, by extracting the text direction value:

```js
var isLtr = editor.lang.dir === 'ltr';

addButtonCommand( 'BidiLtr', lang.ltr, 'bidiltr', bidiCommand( 'ltr' ), isLtr ? 10 : 20 );
addButtonCommand( 'BidiRtl', lang.rtl, 'bidirtl', bidiCommand( 'rtl' ), isLtr ? 20 : 10 );
```